### DISCUSSION TITLE
Upgradeable Smart Contract

### DISCUSSION THREAD

**Discussion by SpringxDSay** [Cyfrin/foundry-full-course-cu#2528] [General]
While writing the upgradeBox function in the UpgradeBox contract, I realised that calling upgradeTo on the proxy address returns an error. But if you control click into the ERC1967Proxy contract, an upgradeToAndCall function will be found contrary to the upgradeTo function used by Patrick. So the address of the newBox will be passed as the first argument and "" as the second

**Comment by EngrPips** [comment]
You can pass an empty byte string as the second argument I think.

**Reply by SpringxDSay** [reply]
that's what I did

**Reply by EngrPips** [reply]
Interesting.
